targets/psa_switch/externs/psa_meter.h

```diff
@@ -0,0 +1,90 @@
+/* Copyright 2020-present Cornell University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * Yunhe Liu (yunheliu@cs.cornell.edu)
+ *
+ */
+
+#ifndef PSA_SWITCH_PSA_METER_H_
+#define PSA_SWITCH_PSA_METER_H_
+
+#include <bm/bm_sim/extern.h>
+#include <bm/bm_sim/meters.h>
+
+namespace bm {
+
+namespace psa {
+
+class PSA_Meter : public bm::ExternType {
+ public:
+  static constexpr p4object_id_t spec_id = 0xfffffffe;
+
+  BM_EXTERN_ATTRIBUTES {
+    BM_EXTERN_ATTRIBUTE_ADD(n_meters);
+    BM_EXTERN_ATTRIBUTE_ADD(type);
+    BM_EXTERN_ATTRIBUTE_ADD(is_direct);
+    BM_EXTERN_ATTRIBUTE_ADD(rate_count);
+  }
+  
+  void init() override {
+    bm::MeterArray::MeterType meter_type;
+    if (type == "bytes") {
+        meter_type = bm::MeterArray::MeterType::BYTES;
+    } else if (type == "packets") {
+        meter_type = bm::MeterArray::MeterType::PACKETS;
+    } else {
+        // TODO: Here Meter is made default as a BYTES meter.
+        // However, what really should been done here is error handling.
+        meter_type = bm::MeterArray::MeterType::BYTES;
+    }
+    _meter = std::unique_ptr<MeterArray>(
+        new MeterArray(get_name() + ".$impl",
+                         spec_id,
+                         meter_type,
+                         rate_count.get<size_t>(),
+                         n_meters.get<size_t>()));
+
+        // Default trTriColor meter rates
+        // 2 packets per second, burst size of 3
+        Meter::rate_config_t committed_rate = {0.000002, 3};
+        // 10 packets per second, burst size of 1
+        Meter::rate_config_t peak_rate = {0.00001, 1};
+        Meter::MeterErrorCode error = _meter->set_rates({committed_rate, peak_rate});
+        if (error != bm::MeterArray::MeterErrorCode::SUCCESS) {
+            // TODO return error handling
+        }
```

Comment:
indentation is wrong

also, not needed. If no rates are configured, execute will always return "green", which I believe is expected

These default values you have picked look very arbitrary